toolkit/tools/imagecustomizerapi/imageFormatType.go

```diff
@@ -0,0 +1,47 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+package imagecustomizerapi
+
+import (
+	"fmt"
+	"slices"
+)
+
+type ImageFormatType string
+
+const (
+	ImageFormatTypeNone     ImageFormatType = ""
+	ImageFormatTypeVhd      ImageFormatType = "vhd"
+	ImageFormatVhdTypeFixed ImageFormatType = "vhd-fixed"
+	ImageFormatTypeVhdx     ImageFormatType = "vhdx"
+	ImageFormatTypeQcow2    ImageFormatType = "qcow2"
+	ImageFormatTypeRaw      ImageFormatType = "raw"
+	ImageFormatTypeIso      ImageFormatType = "iso"
+	ImageFormatTypeCosi     ImageFormatType = "cosi"
+)
+
+// supportedImageFormatTypes is a list of all non-empty image format types
+// defined above.
+var supportedImageFormatTypes = []string{
+	string(ImageFormatTypeVhd),
+	string(ImageFormatVhdTypeFixed),
+	string(ImageFormatTypeVhdx),
+	string(ImageFormatTypeQcow2),
+	string(ImageFormatTypeRaw),
+	string(ImageFormatTypeIso),
+	string(ImageFormatTypeCosi),
+}
+
+func (ft *ImageFormatType) IsValid() error {
```

Comment:
`string` (`ImageFormatType`) is already a reference type. So, you don't need to capture it by pointer.